Fixed strict standards errors, updated for custom taxonomies

badfeather opened the pull request:
I took a stab at fixing the strict standards warnings and updating the plugin to allow for custom taxonomies. It seems to be working correctly in my tests.

Thanks for the plugin!

scribu commented on main.php:
```diff
@@ -18,18 +18,19 @@ function init() {
+			$var->read_cookie();
 		elseif ( apply_filters( 'smarter_nav_set', $set_condition ) )
-			self::set_cookie();
+			$var->set_cookie();
```
Instead of using the `$var = new self();` hack, it would be cleaner to just make all the methods static.

scribu commented on template-tags.php:
```diff
@@ -23,6 +23,7 @@ function referrer_link( $format = '%link', $title = '%title', $sep = '&raquo;',
 }
 
 // Retrieve the category, based on the referrer URL. Useful if you have posts with multiple categories
+// This should be phased out and replaced with get_referrer_term
```
You could start that process by making `get_referrer_category()` call `get_referrer_term()` internally.

badfeather replied:
Ah, good idea.

badfeather commented:
Hey @scribu, I changed the methods in main.php to static as you suggested, and removed the `$var = new self();` hack. In template-tags.php, `get_referrer_category()` is now using `get_referrer_term()`, which opens up the plugin so that it can be used for custom taxonomies.

scribu commented:
Excellent.

scribu commented:
Version 1.4 is now live on wp.org: https://wordpress.org/plugins/smarter-navigation/

badfeather commented:
Sweet! Thanks for pushing it live.

On Fri, Sep 18, 2015 at 6:25 PM, Cristi Burcă <notifications@github.com>
wrote:

> Version 1.4 is now live on wp.org:
> https://wordpress.org/plugins/smarter-navigation/
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/scribu/wp-smarter-navigation/pull/9#issuecomment-141583102>
> .
>



-- 
badfeather.com
